tmail-backend/mailbox/encrypted/pgp-utils/src/main/java/com/linagora/tmail/pgp/ClearEmailContentExtractor.java

```diff
@@ -0,0 +1,114 @@
+package com.linagora.tmail.pgp;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import org.apache.james.jmap.api.model.Preview;
+import org.apache.james.mailbox.model.ParsedAttachment;
+import org.apache.james.mailbox.store.mail.model.impl.MessageParser;
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.util.html.HtmlTextExtractor;
+import org.apache.james.util.mime.MessageContentExtractor;
+
+import com.google.common.base.Preconditions;
+
+public class ClearEmailContentExtractor {
```

Comment:
Why in pgp-utils? I think it should be in mailbox-encrypted-api. If you check the issue again as well, the classes are defined in Scala

You would not need all the following boiler plate code with Scala as well

Reply:
> If you check the issue again as well, the classes are defined in Scala

I see, I just think that is the way Benoit wants to explain object class. I didn't think this class requires scala/java languages implicit.

Reply:
No it is scala syntax :) If you struggle with Scala don't hesitate to ask questions